Tiny PR Adding a note about astro dev run being local in the Variables guide (#2538)
astronomer/docs: learn/airflow-variables.md

astro dev run variables set -j my_json_var '{"key": "value"}'
```

> Note that [`astro dev run`](https://docs.astronomer.io/astro/cli/astro-dev-run) executes Airflow commands in your local Astro CLI scheduler container and cannot be applied to Deployments running on Astro. To set Airflow variables for an Astro Deployment, either create them in the Deployments Airflow UI or via an [environment variable](https://docs.astronomer.io/astro/environment-variables).

Reviewer: ```suggestion
Note that [`astro dev run`](https://docs.astronomer.io/astro/cli/astro-dev-run) executes Airflow commands only in your local Airflow environment and can't be used on Astro Deployments. To set Airflow variables for an Astro Deployment, either create them using the Deployment's Airflow UI or using [Astro environment variables](https://docs.astronomer.io/astro/environment-variables).
```
